RadixWallet/Features/SettingsFeature/Preferences/HiddenAssets/HiddenAssets.swift

```diff
@@ -0,0 +1,225 @@
+			return .none
+
+		case let .setNonFungible(values):
+			state.nonFungible = values
+			return .none
+
+		case let .setPoolUnit(values):
+			state.poolUnit = values
+			return .none
+
+		case let .didUnhideAsset(asset):
+			switch asset {
+			case let .fungible(resourceAddress):
+				state.fungible.removeAll(where: { $0.resourceAddress == resourceAddress })
+			case let .nonFungible(globalId):
+				state.nonFungible.removeAll(where: { $0.token.id == globalId })
+			case let .poolUnit(poolAddress):
+				state.poolUnit.removeAll(where: { $0.details.address == poolAddress })
+			}
+			state.destination = nil
+			return .none
+		}
+	}
+
+	public func reduce(into state: inout State, presentedAction: Destination.Action) -> Effect<Action> {
+		switch presentedAction {
+		case let .unhideAlert(action):
+			switch action {
+			case let .confirmTapped(asset):
+				return .run { send in
+					try await appPreferencesClient.updating { preferences in
+						preferences.assets.unhideAsset(asset: asset)
+					}
+					await send(.internal(.didUnhideAsset(asset)), animation: .default)
+				}
+			case .cancelTapped:
+				state.destination = nil
+				return .none
+			}
+		}
+	}
+
+	private func fungibleEffect(hiddenAssets: [AssetAddress]) -> Effect<Action> {
+		.run { send in
+			let resources = try await onLedgerEntitiesClient.getEntities(addresses: hiddenAssets.fungibleAddresses, metadataKeys: .resourceMetadataKeys).compactMap(\.resource)
+			await send(.internal(.setFungible(resources)))
+		}
+	}
+
+	private func nonFungibleEffect(hiddenAssets: [AssetAddress]) -> Effect<Action> {
+		.run { send in
+			let nonFungibleDetails = try await hiddenAssets.nonFungibleDictionary.parallelMap { resourceAddress, nonFungibleIds in
+				let resource = try await onLedgerEntitiesClient.getResource(resourceAddress)
+				let tokens = try await onLedgerEntitiesClient.getNonFungibleTokenData(.init(resource: resourceAddress, nonFungibleIds: nonFungibleIds))
+				return tokens.map {
+					State.NonFungibleDetails(resource: resource, token: $0)
+				}
+			}
+			.flatMap { $0 }
+
+			await send(.internal(.setNonFungible(nonFungibleDetails)))
+		}
+	}
+
+	private func poolUnitEffect(hiddenAssets: [AssetAddress]) -> Effect<Action> {
+		.run { send in
+			let resourcePools = try await onLedgerEntitiesClient.getEntities(addresses: hiddenAssets.poolUnitAddresses, metadataKeys: .resourceMetadataKeys).compactMap(\.resourcePool)
+			let resources = try await resourcePools.parallelMap {
+				try await onLedgerEntitiesClient.getResource($0.poolUnitResourceAddress)
+			}
+			let poolUnitDetails = try await resources.parallelMap { resource in
+				if let details = try await onLedgerEntitiesClient.getPoolUnitDetails(resource, forAmount: .one) {
+					State.PoolUnitDetails(resource: resource, details: details)
+				} else {
+					nil
+				}
+			}
+			.compactMap { $0 }
+			await send(.internal(.setPoolUnit(poolUnitDetails)))
+		}
+	}
+}
+
+// MARK: - Helpers
+
+private extension [AssetAddress] {
```

Comment:
personal preferences, I dislike `extension [Element]` syntax - since it makes it impossible to search for all Array extensions and all Dictionary extensions individually. But I think I'm the only one to dislike it and we are using it elsewhere.

Reply:
I actually used to add extensions like this as `extension Array where Element == XYZ` before, would you prefer it that way?

Reply:
Yes I prefer that, but then we have fragmentation in the code base. I think this is the most common syntax we use, so maybe best stick to it